### PR TITLE
Enable crb <=> the Optional repo is enabled

### PR DESCRIPTION
**Pull request by pirat89**
Blacklist the CRB repository (as before) iff the Optional repository is not enabled. Otherwise keep the rest on other actors that will handle it as other repositories. In case that the CRB repository is used during the upgrade (can be only when it is not blacklisted), enable the repo on the target system as well during the `FirstBoot` phase.

*Note*: Planned todo in future: the actor is working with this one repo now, but would be nice to make it more robust to be used for any repository task in future. See the TODO msg in code.

### CODE REVIEW

**Comment by pirat89** [comment]
Btw, probably would be better to inform even that when Optional channel is enabled, CRB will be enabled as well and in such case upgrade is not supported (or any error related somehow to the content in those repos)

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/actor.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from leapp.models import UsedTargetRepositories
+from leapp.tags import FirstBootPhaseTag, IPUWorkflowTag
+
+
+class EnableCRBrepo(Actor):
+    """
+    Enable CRB repository when used during the upgrade.
+    """
+    # TODO: replace this actor by more robust one which will enable repositories
+    # based on msg; be careful about the solution as there is similar think
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/enablecrbrepo/actor.py]: typo
```suggestion
    # based on msg; be careful about the solution as there is similar thing
```

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import UsedTargetRepositories
+
+
+CRB_REPOID = 'codeready-builder-for-rhel-8-x86_64-rpms'
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py]: What about moving this to config actor?

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            return True
+    return False
+
+
+def process():
+    if _is_crb_used():
+        try:
+            run(['subscription-manager', 'repos', '--enable', CRB_REPOID])
+        except CalledProcessError as e:
+            api.current_logger().error('Cannot enable CRB repository: %s', str(e))
```
Comment (vojtechsokol) [inline repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py]:
```suggestion
            api.current_logger().error('Cannot enable CRB repository: {}'.format(str(e)))
```

File: repos/system_upgrade/el7toel8/actors/repositoriesblacklist/actor.py [hunk 4]
```diff
@@ -9,13 +9,21 @@ class RepositoriesBlacklist(Actor):
     """
 
     name = 'repositories_blacklist'
-    consumes = ()
+    consumes = (RepositoriesFacts,)
     produces = (RepositoriesBlacklisted,)
     tags = (IPUWorkflowTag, FactsPhaseTag)
 
+    def _is_repo_enabled(self, repoid):
```
Comment (jmikovic) [inline repos/system_upgrade/el7toel8/actors/repositoriesblacklist/actor.py]: shouldn't this function be moved to library?

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py (hunk 2 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py]: Nope as this is not anything related to the config

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py (hunk 3 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/enablecrbrepo/libraries/library.py]: regarding discussion, this is allowed for logging (see various other actors)

File: repos/system_upgrade/el7toel8/actors/enablecrbrepo/actor.py (hunk 1 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/enablecrbrepo/actor.py]: fixed

File: repos/system_upgrade/el7toel8/actors/repositoriesblacklist/actor.py (hunk 4 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/repositoriesblacklist/actor.py]: Not necessary. The actor can be tested as whole component. I am thinking more about shared library to work with repositories completely but this is for following PR.

**Pull request merged**